### PR TITLE
Log request duration (GSI-1220)

### PR DESCRIPTION
**Pull request by TheByronHimes**

### CODE REVIEW

File: tests/integration/test_api.py [hunk 1]
```diff
@@ -92,3 +93,31 @@ async def test_route():
...
+
+    # Get list of the log messages
+    for record in caplog.record_tuples:
+        if record[0] == "ghga_service_commons.api.api":
+            msg = record[2]
+            assert msg.endswith(" ms")
+            msg = msg.removesuffix(" ms")
+
+            duration = msg.rsplit(" ", 1)[1]
+            assert duration.isnumeric
```
Comment (Cito) [inline tests/integration/test_api.py:120]:
```suggestion
            assert re.match(r'GET http://.* "200 OK" - \d+ ms$', record[2])
```

File: src/ghga_service_commons/api/api.py [hunk 2]
```diff
@@ -217,6 +219,29 @@ async def correlation_id_middleware(
         return response
 
 
+async def request_logging_middleware(request: Request, call_next):
+    """Measure and log the amount of time it takes to process the HTTP request."""
+    url = request.url
+    start_time = time.time()
+    response = await call_next(request)
+    duration = round((time.time() - start_time) * 1000)
```
Comment (Cito) [inline src/ghga_service_commons/api/api.py:227]:
```suggestion
    duration = int(round((time.time() - start_time) * 1000))
```

Just to make sure that there is no unnecessary ".0" in the text or JSON representation.

File: src/ghga_service_commons/api/api.py [hunk 3]
```diff
@@ -217,6 +219,29 @@ async def correlation_id_middleware(
...
+    except ValueError:
+        status_phrase = ""
+    msg = f'{request.method} {url} "{response.status_code} {status_phrase}" - {duration} ms'
+    log.info(
+        msg,
+        extra={
+            "method": request.method,
+            "url": str(url),
+            "status_code": response.status_code,
+            "duration": duration,
```
Comment (Cito) [inline src/ghga_service_commons/api/api.py:239]:
```suggestion
            "duration_ms": duration,
```

so that the unit is clear from the json

**Review by Cito — APPROVED** [review]
Some minor suggestions

File: tests/integration/test_api.py (hunk 1 above)
Comment (TheByronHimes) [reply tests/integration/test_api.py]: Thanks for the suggestion. See [e6c0a79](https://github.com/ghga-de/ghga-service-commons/pull/54/commits/e6c0a7964f63418cbe64ebace52dd67b0722a9ac)

**Review by Cito — APPROVED** [review]

**Pull request merged**